stable_baselines3/common/monitor.py

```diff
@@ -38,33 +41,26 @@ def __init__(
     ):
         super(Monitor, self).__init__(env=env)
         self.t_start = time.time()
-        if filename is None:
-            self.file_handler = None
-            self.logger = None
+        if filename:
+            self.results_writer = ResultsWriter(
```

Comment:
Did you check that this method was equivalent to what was there before?

Reply:
Yes, and all the test cases went through.